app/src/main/java/pl/droidsonroids/droidsmap/feature/office/ui/OfficeUiFeatureView.kt

```diff
@@ -84,72 +82,26 @@ class OfficeUiFeatureView(private val activity: MapActivity) : BaseFeatureView<O
         }))
     }
 
-    override fun prepareForRoomTransition() {
-        with(activity.roomImage) {
-            val resources = activity.resources
-            val resourceId = resources.getIdentifier("room_3", "drawable", activity.packageName)
-            val roomImageDrawable = resources.getDrawable(resourceId)
+    override fun onLocationPermissionGranted() {
+        googleMap?.isMyLocationEnabled = true
+    }
 
-            setImageDrawable(roomImageDrawable)
-            layoutParams.width = ((roomImageDrawable).intrinsicWidth * 2.2f).toInt()
-            layoutParams.height = ((roomImageDrawable).intrinsicHeight * 2.2f).toInt()
+    override fun onPerspectiveChanged(active: Boolean) = Unit
 
-            googleMap?.run {
-                with(projection.visibleRegion) {
-                    addMarker(MarkerOptions().position(nearLeft).title("Near left"))
-                    addMarker(MarkerOptions().position(nearRight).title("Near right"))
-                    addMarker(MarkerOptions().position(farLeft).title("Far left"))
-                    addMarker(MarkerOptions().position(farRight).title("Far right"))
-                }
-            }
-        }
-    }
+    override fun requestOffice() = presenter.onRequestOffice()
 
-    private fun createRoomList() {
-        roomsList.add(Room(228f, 144f, 114f, 138f, "skybuds room", R.drawable.room_3))
-        roomsList.add(Room(148f, 204f, 74f, 304f, "server room", R.drawable.room_2))
-        roomsList.add(Room(100f, 168f, 50f, 454f, "server room2", R.drawable.room_2a))
-        roomsList.add(Room(164f, 116f, 174f, 456f, "room 1", R.drawable.room_1))
-        roomsList.add(Room(126f, 142f, 257f, 273f, "room 4", R.drawable.room_4))
-        roomsList.add(Room(166f, 202f, 301f, 607f, "room m", R.drawable.room_m))
-        roomsList.add(Room(144f, 202f, 448f, 607f, "room 5", R.drawable.room_5))
-        roomsList.add(Room(256f, 204f, 562f, 304f, "room 6", R.drawable.room_6))
-        roomsList.add(Room(130f, 70f, 377f, 371f, "wall 1", R.drawable.wall_1))
-        roomsList.add(Room(146f, 202f, 585f, 607f, "room_7", R.drawable.room_7))
-        roomsList.add(Room(194f, 406f, 759f, 203f, "room_fun", R.drawable.room_fun))
-        roomsList.add(Room(206f, 202f, 753f, 607f, "room_9", R.drawable.room_9))
+    override fun setMapPanningConstraints(uiModel: OfficeUiModel) {
+        appointUiTaskIfMapNull { performSetMapPanningConstraints(uiModel) }
     }
 
-    private fun checkLocationPermission() {
-        if (ContextCompat.checkSelfPermission(activity, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
-            googleMap?.isMyLocationEnabled = true
+    private fun appointUiTaskIfMapNull(function: () -> Any) {
+        if (googleMap == null) {
+            UiCommandInvoker.queueInvokement { function }
```

Comment:
Ja dalej mam wątpliwości, czy my to dobrze robimy z tym UI commandem, ale to jeszcze przegadamy.